### PR TITLE
✨ Add Content Calendar view with filters

### PR DESCRIPTION
**Pull request by brandenwagner**
Introducing a full Content Calendar experience for Posts, moving beyond scheduled-only items.
image

What’s included
 - Multi-filter support to the calendar
 - Same filter experience across the rest of the posts pages
 - Persisted filters in URL search params so calendar state is shareable and reload-safe.
- Added status legend + visual styling per post status in the grid.
- Added flexible date fallback logic for calendar placement:
- Drafts prefer updated_at, then created_at, then published_at.
- Other statuses prefer published_at, then updated_at, then created_at.
- Added sort support in calendar utilities (published_at asc/desc, updated_at desc).
- Updated unit tests to cover new status/date/sort behavior and missing-date edge cases.

Please take a minute to explain the change you're making:
- **Why are you making it?**
  - Its been discussed before on the forums but no one (to my knowledge) has started working on this feature. It helps me see the schedule and keep things on track. Especially when im working on a series by tags
- What does it do?
  - Its a calendar display of posts 
- Why is this something Ghost users or developers need?
  - Its a visual interpretation of the existing post listing pages.

Please check your PR against these items:
- [x] I've read and followed the [Contributor Guide](https://github.com/TryGhost/Ghost/blob/main/.github/CONTRIBUTING.md)
- [x] I've explained my change
- [x] I've written an automated test to prove my change works

---

> [!NOTE]
> **Medium Risk**
> Adds a new routed UI surface that relies on Admin API filtering/sorting and timezone/date fallbacks, which could cause subtle correctness/performance issues with large post sets or unexpected date data. Changes are otherwise isolated (types, navigation link, and dev compose healthcheck).
> 
> **Overview**
> Introduces a new `posts/calendar` route and a corresponding **Content Calendar** view that renders posts in a month grid, supports type/visibility/author/tag/sort filters persisted in URL search params, and shows status-based styling/legend with month navigation.
> 
> Adds calendar utilities to place and sort posts by timezone-aware day keys with date-field fallback rules (drafts prefer `updated_at` then `created_at`; others prefer `published_at`), plus unit tests covering timezone, ordering, and missing-date edge cases. Also updates the Posts API `Post` type to include `created_at`/`updated_at`, adds a Posts sidebar subnav link for Calendar, and switches the dev MySQL healthcheck to `mysqladmin ping`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 3ffb277cb926337796e0f256ecf3c258df119e30. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: apps/posts/src/views/ContentCalendar/content-calendar.tsx [hunk 1]
```diff
@@ -0,0 +1,702 @@
...
+                                                    >
+                                                        {style.label}
+                                                    </span>
+                                                );
+                                            })}
+                                        </div>
+                                    )}
+                                    <MonthNavigation monthLabel={monthLabel} monthOffset={monthOffset} setMonthOffset={setMonthOffset} />
+                                </div>
+                                {posts.length === 0 ? (
```
[the remark this answers is not in the document]
Comment (brandenwagner) [reply apps/posts/src/views/ContentCalendar/content-calendar.tsx]: Fixed in [504ed94](https://github.com/TryGhost/Ghost/pull/26430/commits/504ed94ff06dcb133d4afb3ef13c24489817c329)

File: apps/posts/src/views/ContentCalendar/content-calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+const getCalendarDateRangeFilter = (month: CalendarMonth): string => {
+    const rangeStart = new Date(Date.UTC(month.year, month.month - 2, 1, 0, 0, 0, 0)).toISOString();
+    const rangeEnd = new Date(Date.UTC(month.year, month.month + 1, 0, 23, 59, 59, 999)).toISOString();
+
+    const rangeForField = (field: string) => {
+        return `${field}:>='${rangeStart}'+${field}:<='${rangeEnd}'`;
+    };
+
+    return `(${rangeForField('published_at')},${rangeForField('updated_at')},${rangeForField('created_at')})`;
+};
```
[the remark this answers is not in the document]
Comment (brandenwagner) [reply apps/posts/src/views/ContentCalendar/content-calendar.tsx]: I think this is wrong. The worst case for trailing cells: a month that starts on Saturday (offset 6) with 31 days (e.g. August 2026).

Cells 0–5: last 6 days of July
Cells 6–36: August 1–31
Cells 37–41: September 1–5
With month.month = 8 (August), rangeEnd = Date.UTC(2026, 9, 0, ...) → day 0 of October = September 30. That covers September 1–5 just fine.

The maximum trailing day shown in the next month is 42 - firstDayOffset - daysInCurrentMonth days into the next month. That's at most 42 - 0 - 28 = 14 (for a 28-day February starting on Sunday), which is March 14 at most. rangeEnd at March 31 covers this completely.

The current rangeEnd is already correct. Date.UTC(month.year, month.month + 1, 0) gives the last day of month.month + 1 in 1-based terms (i.e. the next month after the displayed month), which is always beyond the furthest trailing cell that can appear in the 42-cell grid.

Can someone else double check me or is there another way to render it?

**Comment by 9larsons** [comment]
@brandenwagner  Thanks for this, and apologies for the long silence. I want to be upfront about where this stands: the implementation is genuinely well done (right stack; React in apps/posts with shade components and admin-x-framework with unit tests on the date logic).

That said, a content calendar is a whole new product surface in admin, and that's a decision our product/design team has to own before any PR (our's or a contributor's) can land it. We're going to discuss whether this is a feature we want to take on, and I'll report back here once there's a direction.

If the answer is yes, the likely path would be: gating it behind a labs flag initially, replacing the limit:'all' queries with month-windowed fetches, and sanitizing the URL-param→NQL filter interpolation plus enabling maintainer edits on the branch (or we'd take it over). 

Any chance you have a screenshot still? The image above isn't rendering. That or a brief video would be very helpful for consideration.

**Comment by brandenwagner** [comment]
@9larsons  here is a screenshot. calendar
I can work on putting together a video later

Totally get product team owning this, so no rush. I have it on my own fork and figured id share it. Its been super useful to visualize when my posts are going out.

**Comment by 9larsons** [comment]
Hey @brandenwagner, thanks for the submission! There's a lot of interest in this idea, but there's a few product hurdles I think we need to tackle before something like this can be fully realized. 

The calendar is driven off scheduled posts, and in Ghost "scheduled" means "ready to publish at this date", not "planned for this date." A real editorial calendar is a living view of planned/WIP work that moves around as you figure out what's running when and we don't model that today. Driving the surface off scheduled-posts conflates planning with publishing, which is fine for your workflow but actively risky for publishers who'd start treating "set a date" as planning. This is an active conversation we're having.

Thanks for the contribution and the patience - going to close this out for now.

**Comment by brandenwagner** [comment]
Thanks for the info! I totally get that and I wasn't thinking about that workflow for others at all. Appreciate the quality you are putting into it. If you want help on I'm happy to contribute.
